File: app/src/main/java/com/infy/dagger2implementation/di/module/ActivityBindingModule.java
package com.infy.dagger2implementation.di.module;

import com.infy.dagger2implementation.activity.MainActivity;
import com.infy.dagger2implementation.base.BaseActivity;
import com.infy.dagger2implementation.base.BaseFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityBindingModule {
    @ContributesAndroidInjector(modules = ActivityModule.class)
    abstract BaseActivity baseActivity();

    @ContributesAndroidInjector
    abstract MainActivity mainActivity();

    @ContributesAndroidInjector
    abstract BaseFragment provideBaseFragment();
}

File: app/src/main/java/com/infy/dagger2kotlin/di/module/ActivityBindingModule.kt
package com.infy.dagger2kotlin.di.module

import com.infy.dagger2kotlin.activity.MainActivity
import com.infy.dagger2kotlin.base.BaseActivity
import com.infy.dagger2kotlin.base.BaseFragment
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityBindingModule {
    @ContributesAndroidInjector(modules = arrayOf(ActivityModule::class))
     abstract fun baseActivity(): BaseActivity

    @ContributesAndroidInjector
     abstract fun mainActivity(): MainActivity

    @ContributesAndroidInjector
     abstract fun provideBaseFragment(): BaseFragment
}
